@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -2232,32 +2242,27 @@ export class TextElementsRenderer {
         return this.m_overloaded;
     }
 
-    private checkStartFadeIn(
-        renderState: RenderState | undefined,
-        frameNumber: number,
-        time: number,
-        forceFadeIn = false
-    ): boolean {
-        let fadeAnimationStarted = false;
-
-        if (renderState !== undefined) {
-            // Fade-in after skipping rendering during movement
-            if (
-                forceFadeIn ||
-                renderState.state === FadingState.Undefined ||
-                renderState.lastFrameNumber < frameNumber - 1
-            ) {
-                this.startFadeIn(renderState, frameNumber, time);
-            }
-
-            if (renderState.isFading()) {
-                this.updateFading(renderState, time);
-                fadeAnimationStarted = true;
-            }
-
-            renderState.lastFrameNumber = frameNumber;
+    // if not fading in fade in, if fading out reverse
+    private ensureFadeIn(renderState: RenderState, frameNumber: number, time: number) {
```

Comment:
The function named ensureFadeIn returns false if render state is FadedIn and true otherwise. And that is completely non-intuitive.
I would prefer to preserve that variable ```fadeAnimationStarted``` to keep awareness on what's going on.